### DISCUSSION TITLE
how to get credentials for "Write" methods

### DISCUSSION THREAD

**Discussion by Hamza-Shewa** [WalletConnect/walletconnect-monorepo#3826] [Developer Support]
I'm having a problem finding credentials that are required by my abi function, I'm trying to write to the chain, and Apparently there's no way to get the credentials throughout this package! is there's an alternative? 
this is my function:

  Future<String> buyDappOnPinky(
    BigInt _id, {
    required _i1.Credentials credentials,
    _i1.Transaction? transaction,
  }) async {
    final function = self.abi.functions[2];
    assert(checkSignature(function, expected));
    final params = [_id];
    return write(
      credentials,
      transaction,
      function,
      params,
    );
  }

**Comment by glitch-txs** [comment]
Hi this issue lacks information, please go through our [guide](https://github.com/orgs/WalletConnect/discussions/3457) and provide more information. Thanks!
